apollo-core/src/main/java/com/ctrip/framework/apollo/core/utils/DeferredLogger.java

```diff
@@ -0,0 +1,418 @@
+package com.ctrip.framework.apollo.core.utils;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import org.slf4j.Logger;
+import org.slf4j.Marker;
+
+/**
+ * Only the following methods implement the delayed log function
+ * <pre class="code">
+ *  public void debug(String s);
+ *  public void debug(String s, Object o);
+ *  public void debug(String s, Object... objects);
+ *  public void debug(String s, Object o, Object o1);
+ *  public void info(String s);
+ *  public void info(String s, Object o);
+ *  public void info(String s, Object... objects);
+ *  public void info(String s, Object o, Object o1);
+ *  public void warn(String s);
+ *  public void warn(String s, Object o);
+ *  public void warn(String s, Object... objects);
+ *  public void warn(String s, Object o, Object o1);
+ *  public void error(String s);
+ *  public void error(String s, Throwable throwable);
+ * </pre>
+ *
+ * @author kl (http://kailing.pub)
+ * @since 2021/5/19
+ */
+public class DeferredLogger implements Logger {
+
+  private final Logger logger;
+  /**
+   * Delay the opening state of the log -1: Uninitialized 0: Disable 1: Enabled
+   */
+  private static final AtomicInteger state = new AtomicInteger(-1);
+
+  public static void enable() {
+    state.compareAndSet(-1, 1);
+  }
+
+  public static void disable() {
+    state.set(0);
+  }
+
+  public static boolean isEnabled() {
+    return state.get() == 1;
+  }
+
+  public static void replayTo(){
+    DeferredLogCache.replayTo();
```

Comment:
How about we disable the state here so that DeferredLogCache is simply a cache which doesn't need to care whether the deferred logger is enabled or not?
```suggestion
    disable();
    DeferredLogCache.replayTo();
```

Reply:
It makes sense, my thinking has been stuck on `DeferredLogUtils`